nilearn/decoding/tests/test_decoder.py

```diff
@@ -1144,6 +1156,37 @@ def test_decoder_screening_percentile_surface(perc, _make_surface_class_data):
         assert model.screening_percentile_ == perc
 
 
+@pytest.mark.parametrize("screening_percentile", [30, 80, 100])
+def test_decoder_screening_percentile_adjustment_surface(
+    _make_surface_mask, _make_surface_class_data, screening_percentile
+):
+    mask = _make_surface_mask()
+    img, y = _make_surface_class_data()
+    mask_n_vertices = get_mask_volume(mask)
+    mesh_n_vertices = img.mesh.n_vertices
+    mask_to_mesh_ratio = (mask_n_vertices / mesh_n_vertices) * 100
+    decoder = Decoder(
+        mask=mask,
+        param_grid={"C": [0.01, 0.1]},
+        cv=3,
+        screening_percentile=screening_percentile,
+    )
+    with pytest.warns(UserWarning, match="Consider raising"):
+        decoder.fit(img, y)
+    adjusted = decoder.screening_percentile_
+    if (
+        screening_percentile == 100
+        or mask_to_mesh_ratio < screening_percentile
```

Comment:
I guess that you could simply replace both conditions by  `mask_to_mesh_ratio =< screening_percentile`

Reply:
oops yes indeed